src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs

```diff
@@ -162,6 +162,18 @@ public NuGetVersion Version
         /// <remarks>Optional. This is normally set for internal use only.</remarks>
         public ProjectRestoreMetadata RestoreMetadata { get; set; }
 
+        /// <summary>
+        /// A unified flag to help users manage their SDK warning levels. Example: 9.0.100
+        /// When introducing a new warning or error use this property to
+        /// allow users to tell the sdk "treat me as if I were SDK x.y.z" and manage breaking changes
+        /// </summary>
+        public NuGetVersion SdkAnalysisLevel { get; set; }
+
+        /// <summary>
+        /// Indicates that Microsoft.NET.Sdk is being used.
+        /// </summary>
+        public bool UsingMicrosoftNETSdk { get; set; } = true;
```

Comment:
I've commented on several files how when we're reading the property from the project, we need to handle the property not being set as `false`, and explicitly set this value in that case.

An alternative is to let this bool keep its `false` default, and instead in `JsonPackageSpecReader.cs` to set the property value to `true` when the json doesn't contain the property. That way it only needs to be done in one place, not in every place we convert a project into a PackageSpec.

Reply:
How are you going to use UsingMicrosoftNETSdk? 
Why in particular do we need it?

Reply:
It'll be used to tell the difference between an SDK style project using an older version of the .NET SDK before they added SdkAnalysisLevel, and non-SDK style (legacy) project files, where we can make different decisions about default values.

Reply:
What would those defaults be?

I'm trying to truth table that:

| SdkAnalysisLevel | UsingMicrosoftNETSdk | Explanation | 
| AnyValue | true | Happy case. Respect SDKAnalysisLevel |
| | true | ??? |
| | false | ??? |
| Any Value | false | ??? |

Reply:
For NuGetAudit related changes, it would be:

|SdkAnalysisLevel|UsingMicrosoftNETSdk|Behaviour|
|--|--|--|
|Set|N/A| Set feature defaults per SdkAnalysisLevel|
|Not set|`true`|Assume SdkAnalysisLevel 8.0|
|Not set|`false` or not set|Assume the highest SdkAnalysisLevel each feature changes defaults for|